OpenRA.Game/WorldUtils.cs

```diff
@@ -41,6 +41,29 @@ public static IEnumerable<Actor> FindActorsInCircle(this World world, WPos origi
 			}
 		}
 
+		public static bool ContainsTemporaryBlocker(this World world, CPos cell, Actor ignoreActor = null)
+		{
+			var temporaryBlockers = world.ActorMap.GetActorsAt(cell);
+			if (temporaryBlockers == null)
+				return false;
+
+			foreach (var temporaryBlocker in temporaryBlockers)
+			{
+				if (temporaryBlocker == ignoreActor)
+					return false;
```

Comment:
This needs to be a continue, not a return.  There may be another blocking actor in the cell!